contracts/access/ownable/OwnableInternal.sol

```diff
@@ -26,22 +25,25 @@ abstract contract OwnableInternal is IOwnableInternal {
         return OwnableStorage.layout().owner;
     }
 
-    function _transitiveOwner() internal view virtual returns (address) {
-        address owner = _owner();
+    function _transitiveOwner() internal view virtual returns (address owner) {
+        owner = _owner();
 
         while (owner.isContract()) {
             try IERC173(owner).owner() returns (address transitiveOwner) {
                 owner = transitiveOwner;
             } catch {
-                return owner;
+                break;
             }
         }
-
-        return owner;
     }
 
     function _transferOwnership(address account) internal virtual {
-        OwnableStorage.layout().setOwner(account);
-        emit OwnershipTransferred(msg.sender, account);
+        _setOwner(account);
+    }
+
+    function _setOwner(address account) internal virtual {
+        OwnableStorage.Layout storage l = OwnableStorage.layout();
+        emit OwnershipTransferred(l.owner, account);
```

Comment:
Should the event be emitted after the state change?

Reply:
If this were a large, complex function, it might be worth it to store `l.owner` in stack so that it can be emitted in the event at the end.  But it's clear enough what's happening here without adding the stack variable.